src/librustc/session/mod.rs

```diff
@@ -176,14 +176,15 @@ impl Session {
     pub fn abort_if_errors(&self) {
         self.diagnostic().abort_if_errors();
     }
-    pub fn abort_if_new_errors<F>(&self, mut f: F)
-        where F: FnMut()
+    pub fn abort_if_new_errors<F, T>(&self, f: F) -> T
```

Comment:
I have another PR which removes a bunch of `abort_if_new_errors` calls, and plan to do more there. So, I'll do a rename, but later, if that is OK.